cabot/metricsapp/api/grafana.py

```diff
@@ -0,0 +1,195 @@
+import json
+import logging
+import requests
+import urlparse
+from django.core.exceptions import ValidationError
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_time_range(timestring):
+    """
+    Find the number of digits a time string corresponds to.
+    :param timestring: time string in the format "3h" or "1d" or "10m"
+    :return: integer number of minutes
+    """
+    time_range = None
+    time_digits = int(filter(str.isdigit, timestring))
+
+    # Days
+    if 'd' in timestring:
+        time_range = 1440 * time_digits
+
+    # Hours
+    elif 'h' in timestring:
+        time_range = 60 * time_digits
+
+    # Minutes
+    elif 'm' in timestring:
+        time_range = time_digits
+
+    return time_range
+
+
+def get_grafana_auth_headers(api_key):
+    """
+    Returns request headers for accessing the Grafana API with an API key
+    :param api_key: the API key
+    :return: dictionary of relevant headers
+    """
+    return {'Authorization': 'Bearer {}'.format(api_key)}
+
+
+def get_dashboards(url, api_key):
+    """
+    Get data for all Grafana dashboards
+    :param url: url for the Grafana instance
+    :param api_key: API key for the Grafana instance
+    :return: api response containing dashboard id, title, uri, etc.
+    """
+    response = requests.get(urlparse.urljoin(url, 'api/search'), headers=get_grafana_auth_headers(api_key))
```

Comment:
Another advantage - in the future if we migrate to a different auth scheme, we don't have to update the invocation everywhere.

We can modify code in one place.